frontend/Streamlit_GUI.py

```diff
@@ -0,0 +1,68 @@
+import streamlit
+from PIL import Image
+import os
+import requests
+
+
+streamlit.header("Steel surface defect analyser")
+number = streamlit.slider("Pick one of the 100 sample images")
+
+
+curr_path = os.getcwd()
+data_subdir_path = "/Dataset-trafo"
+data_path = curr_path + data_subdir_path
+
+list_namesg = []
+list_namesd = []
+list_samples = []
+for f in os.listdir(data_path + "/good"):
+    list_namesg.append("/good/" + f)
+for f in os.listdir(data_path + "/defect"):
+    list_namesd.append("/defect/" + f)
+for i in range(50):
+    list_samples.append(list_namesg[i])
+    list_samples.append(list_namesd[i])
+data2 = Image.open(data_path + list_samples[number])
+
+data = streamlit.file_uploader("Or upload own surface image", type="png")
+
+
+own_data = streamlit.toggle("Use own surface image")
+
+if (data or data2) is not None:
+    temp_save_path = None
+    if data is not None and own_data:
+        test = Image.open(data)
```

Comment:
I wouldn't keep the file open the whole time here. You could rather load it in memory, prefarably using 
```python
with Image.open(data) as img: 
    ...
```
and later use the in-memory image to be sent to the fastapi endpoint once the `Analyze` button is clicked. It can be that you need to adapt the definition of the endpoint then slightly.